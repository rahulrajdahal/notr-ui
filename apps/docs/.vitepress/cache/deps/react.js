import {
  require_react
} from "./chunk-UUF3TEGA.js";
import "./chunk-5WRI5ZAA.js";
export default require_react();
//# sourceMappingURL=react.js.map
