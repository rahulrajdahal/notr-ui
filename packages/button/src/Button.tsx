import { css } from "@emotion/react";
import { colors } from "@rahulrajdahal/notr-core";
import React from "react";

const baseStyles = css`
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    font-weight: 500;
    cursor: not-allowed;
  }
`;

const variants = {
  fill: css`
    border-radius: 0.8125rem;
    padding: 1.31rem;

    &:disabled {
      background-color: var(--color-gray-400) !important;
      color: #ffffff;
    }
  `,
  outline: css`
    border: 1px solid var(--color-primary-600);
    border-radius: 0.8125rem;
    padding: 1.31rem;

    &:disabled {
      border-color: var(--color-gray-400);
      color: var(--color-gray-400) !important;
    }
  `,
  text: css`
    &:hover {
      text-decoration: underline;
    }
    &:disabled {
      color: var(--color-gray-400) !important;
    }
  `,
};

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: keyof typeof variants;
  background?: string;
  color?: string;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      variant = "fill",
      disabled = false,
      // background = variant === "fill" && !disabled
      //   ? colors.primary[600]
      //   : undefined,
      color = variant === "fill" ? "#ffffff" : colors.primary[600],
      className = "",
      children,
      ...props
    },
    ref
  ) => {
    return (
      <button
        ref={ref}
        style={
          {
            backgroundColor:
              variant === "fill" && !disabled ? colors.primary[600] : undefined,
            color,
            "--color-primary-600": colors.primary[600],
            "--color-gray-400": colors.gray[400],
          } as React.CSSProperties
        }
        css={css`
          ${baseStyles} ${variants[variant]}
        `}
        className={className}
        disabled={disabled}
        aria-disabled={disabled}
        {...props}
      >
        {children}
      </button>
    );
  }
);

Button.displayName = "Button";

export default Button;
