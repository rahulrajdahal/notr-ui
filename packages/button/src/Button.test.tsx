import { createSerializer } from "@emotion/jest";
import { colors } from "@rahulrajdahal/notr-core";
import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  it("renders with default props and styles", () => {
    // const user = userEvent.setup();

    render(<Button>Notr Button</Button>);
    const button = screen.getByRole("button", { name: /notr button/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`--color-primary-600: ${colors.primary[600]}`);
    expect(button).toHaveStyle(`--color-gray-400: ${colors.gray[400]}`);
    expect(button).toHaveStyle(`background-color: ${colors.primary[600]}`);
    expect(button).toHaveStyle("padding: 1.31rem");
    expect(button).toHaveStyle("color: #ffffff");
    expect(button).toHaveStyle("border-radius: 0.8125rem");
    expect(button).toHaveStyle("font-weight: 700");
    expect(button).toHaveStyle("cursor: default");

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();

    // await user.hover(button);
    // expect(button).toHaveStyle("cursor: pointer");
  });

  it("renders fill variant styles", () => {
    render(<Button variant="fill">Notr Button Filled</Button>);
    const button = screen.getByRole("button", {
      name: /notr button filled/i,
    });
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`--color-primary-600: ${colors.primary[600]}`);
    expect(button).toHaveStyle(`background-color: ${colors.primary[600]}`);
    expect(button).toHaveStyle("color: #ffffff");
    expect(button).toHaveStyle("font-weight: 700");
    expect(button).toHaveStyle("padding: 1.31rem");
    expect(button).toHaveStyle("border-radius: 0.8125rem");

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();
  });

  it("renders outline variant styles", () => {
    render(<Button variant="outline">Notr Button Outline</Button>);
    let button = screen.getByRole("button", {
      name: /notr button outline/i,
    });
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`--color-primary-600: ${colors.primary[600]}`);
    expect(button).toHaveStyle(`color: ${colors.primary[600]}`);
    expect(button).toHaveStyle("border: 1px solid var(--color-primary-600)");
    expect(button).toHaveStyle("font-weight: 700");
    expect(button).toHaveStyle("padding: 1.31rem");
    expect(button).toHaveStyle("border-radius: 0.8125rem");

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();

    render(
      <Button variant="outline" disabled={true}>
        Notr Button Outline Disabled
      </Button>
    );

    button = screen.getByRole("button", {
      name: /notr button outline disabled/i,
    });
    expect(button).toHaveStyle("border-color: var(--color-gray-400)");
  });

  it("renders text variant styles", () => {
    render(<Button variant="text">Notr Button Text</Button>);
    let button = screen.getByRole("button", {
      name: /notr button text/i,
    });
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`--color-primary-600: ${colors.primary[600]}`);
    expect(button).toHaveStyle(`color: ${colors.primary[600]}`);
    expect(button).toHaveStyle("font-weight: 700");

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();
  });

  it("renders with disabled props and styles", () => {
    render(<Button disabled>Notr Button Disabled</Button>);
    const button = screen.getByRole("button", {
      name: /notr button disabled/i,
    });
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`background-color: var(--color-gray-400)`);
    expect(button).toHaveStyle("cursor:not-allowed");
    expect(button).toHaveStyle("font-weight: 500");
  });
});
