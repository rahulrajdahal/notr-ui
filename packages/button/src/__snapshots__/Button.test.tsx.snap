// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > renders fill variant styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  background-color: var(--color-gray-400)!important;
  color: #ffffff;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  background-color: var(--color-gray-400)!important;
  color: #ffffff;
}

<button
  aria-disabled="false"
  class="emotion-0"
  style="background-color: rgb(53, 99, 222); color: rgb(255, 255, 255); --color-primary-600: #3563de; --color-gray-400: #2a292e;"
>
  Notr Button Filled
</button>
`;

exports[`Button Component > renders outline variant styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border: 1px solid var(--color-primary-600);
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  border-color: var(--color-gray-400);
  color: var(--color-gray-400)!important;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border: 1px solid var(--color-primary-600);
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  border-color: var(--color-gray-400);
  color: var(--color-gray-400)!important;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border: 1px solid var(--color-primary-600);
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  border-color: var(--color-gray-400);
  color: var(--color-gray-400)!important;
}

<button
  aria-disabled="false"
  class="emotion-0"
  style="color: rgb(53, 99, 222); --color-primary-600: #3563de; --color-gray-400: #2a292e;"
>
  Notr Button Outline
</button>
`;

exports[`Button Component > renders text variant styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:disabled {
  color: var(--color-gray-400)!important;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:disabled {
  color: var(--color-gray-400)!important;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:disabled {
  color: var(--color-gray-400)!important;
}

.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:disabled {
  color: var(--color-gray-400)!important;
}

<button
  aria-disabled="false"
  class="emotion-0"
  style="color: rgb(53, 99, 222); --color-primary-600: #3563de; --color-gray-400: #2a292e;"
>
  Notr Button Text
</button>
`;

exports[`Button Component > renders with default props and styles 1`] = `
.emotion-0 {
  outline: none;
  border: none;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.125rem;
  border-radius: 0.8125rem;
  padding: 1.31rem;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:disabled {
  font-weight: 500;
  cursor: not-allowed;
}

.emotion-0:disabled {
  background-color: var(--color-gray-400)!important;
  color: #ffffff;
}

<button
  aria-disabled="false"
  class="emotion-0"
  style="background-color: rgb(53, 99, 222); color: rgb(255, 255, 255); --color-primary-600: #3563de; --color-gray-400: #2a292e;"
>
  Notr Button
</button>
`;
